### PR TITLE
EWARM-12658: Add CMSIS-R Profile to ICCARM

### PR DESCRIPTION
**Pull request by RobinKastberg**
* Created cmsis_iccarm.h and moved generic intrinsics there
* Removed intrinsics which moved to cmsis_iccarm.h from
  cmsis_iccarm_a.h and cmsis_iccarm_m.h
* Removed support for __ICCARM_INTRINCS_VERSION__ < 2 since that was
  very hard to keep while doing this refactoring. It was supposed
  to happen when we went to CMSIS_6.
* As a side-effect, this refactoring will fix
  __CLREX and __STREX bugs fixed for Cortex-M that wasn't backported to
  Cortex-A.

### CODE REVIEW

File: CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h [hunk 1]
```diff
@@ -0,0 +1,171 @@
+/**************************************************************************//**
+ * @file     cmsis_armclang_r.h
+ * @brief    CMSIS compiler armclang (Arm Compiler 6) header file
+ * @version  V6.0.0
+ * @date     04. December 2024
+ ******************************************************************************/
+/*
+ * Copyright (c) 2009-2023 Arm Limited. All rights reserved.
```
Comment (JonatanAntoni) [inline CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h:8]: Would you want add IAR copyright here as well?

Comment (GuentherMartin) [reply CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h:8]: Please change date. 04. December 2024 is not yet reached.
Please change Copyright year to 2009-2024

**Review by GuentherMartin — APPROVED** [review]

File: CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h (hunk 1 above)
Comment (RobinKastberg) [reply CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h:8]: Things went a little bit quick here, just wanted to get a draft out of the door.

> Would you want add IAR copyright here as well?

Sure 

> Please change date. 04. December 2024 is not yet reached.

Hm. This is actually based on https://github.com/ARM-software/CMSIS_6/blob/main/CMSIS/Core/Include/r-profile/cmsis_armclang_r.h and should be fixed there as well.

Comment (JonatanAntoni) [reply CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h:8]: Probably this is a typo, and the year actually was 2023 not 2024.

Comment (JonatanAntoni) [reply CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h:8]: @GuentherMartin, didn't we remove these (dublicated info) from the file headers, anyway? I.e., only the copyright shall be kept, but no file versions and dates?

Comment (GuentherMartin) [reply CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h:8]: I would suggest to remove date and version items from the header in upcoming changes.

Comment (GuentherMartin) [reply CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h:8]: The file header has been removed from all include file except:
  cmsis_armclamg.h, cmsis_gcc.h, cmsis_clang.h
  cmsis_armclamg_r.h, cmsis_gcc_r.h, cmsis_clang_r.h

I do not know why we kept the file headers in these files. We should remove them.

Comment (JonatanAntoni) [reply CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h:8]: I think that happened during the refactoring/reordering of the files. So, let's remove the headers in upcoming changes, as you suggested.

Comment (RobinKastberg) [reply CMSIS/Core/Include/r-profile/cmsis_iccarm_r.h]: Fixed!

**Review by GuentherMartin — APPROVED** [review]
LGTM

**Comment by RobinKastberg** [comment]
I have done some manual testing, checking for regressions with Cortex-M and A. 
And it seems to work well.,
However the lack of automated testsuites and a spec for Cortex-R makes me a little bit anxious.
I have asked for IAR internal reviews but vacation period is upon us, so it could take a while.

**Pull request merged**